### PR TITLE
Add stable hashCode() calculation to PrimitiveSerialDescriptor

### LINKED ISSUE #2135: Make PrimitiveSerialDescriptor.hashCode() stable
**Describe the bug**
`PrimitiveSerialDescriptor` class doesn't have custom hashCode() implemented, so, it can change between different JVMs vendors, versions, servers, between different app run etc.
Plus, we can change hashCode generation strategy (`-XX:hashCode` JVM option) – this will change hashCode value too.

I think it is good idea to add stable hashCode() implementation to `PrimitiveSerialDescriptor`

### PR DESCRIPTION
**Pull request by vgv**
Fix #2135

### CODE REVIEW

**Comment by sandwwraith** [comment]
You're right; all descriptors should have meaningful equals. Can you please also add a test?

**Comment by vgv** [comment]
I've added tests

File: core/commonTest/src/kotlinx/serialization/internal/PrimitiveSerialDescriptorTest.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package kotlinx.serialization.internal
```
Comment (sandwwraith) [inline core/commonTest/src/kotlinx/serialization/internal/PrimitiveSerialDescriptorTest.kt:1]: We don't usually put tests in the `internal` package, as it is not descriptive. There are already `SerialDescriptorBuilderTest` and `SerialDescriptorEqualityTest` that look like a proper fit for these test functions. Alternatively, you can just put `PrimitiveSerialDescriptorTest` near them.

**Review by sandwwraith — CHANGES_REQUESTED** [review]
Thank you! Just move tests out of internal package, please

File: core/commonTest/src/kotlinx/serialization/internal/PrimitiveSerialDescriptorTest.kt (hunk 1 above)
Comment (vgv) [reply core/commonTest/src/kotlinx/serialization/internal/PrimitiveSerialDescriptorTest.kt]: I've checked `SerialDescriptorBuilderTest` and `SerialDescriptorEqualityTest` 
For me, it seems that separate `PrimitiveSerialDescriptorTest` is better than adding test into one of these classes, so I just moved `PrimitiveSerialDescriptorTest` out of `internal` package.

**Review by sandwwraith — APPROVED** [review]

**Comment by sandwwraith** [comment]
This should have been opened against `dev`, but I haven't noticed :( Cherry-picked from master to dev manually.

**Comment by qwwdfsad** [comment]
It's better to merge master into dev instead to avoid duplicate commits while still being in the same repository state eventually

**Pull request merged**